### PR TITLE
Add docs for aws settings and configs

### PR DESCRIPTION
**Pull request by dkatzz**
Related to: https://github.com/civiform/civiform/issues/5078

### CODE REVIEW

File: docs/it-manual/sre-playbook/monitoring.md [hunk 1]
```diff
@@ -57,3 +60,68 @@ To import this pre-built dashboard:
...
+## Additional AWS monitoring
+
+In addition to the server metrics provided by Prometheus, there are some additional places within AWS you can go to see metrics.
+
+### CloudWatch
+
+#### Dashboards
+
+CloudWatch has some default dashboards that allow you to see graphs with metrics on different parts of the deployment. Not all of these are relevant, but these can be helpful in seeing CPU utilization in RDS (the database) and Elastic Container Service, as well as requests to the Application ELB (load balancer), metrics about S3, etc.
+![](<../../.gitbook/assets/aws-cloudwatch-dashboards.png>)
```
Comment (bion) [inline docs/it-manual/sre-playbook/monitoring.md:73]: **nitpick:** Some feedback I got on the existing docs is that they're not quite as beginner friendly as they could be, so these suggestions are mostly in service of remedying that.
```suggestion
[CloudWatch](https://aws.amazon.com/cloudwatch/) has some default dashboards that allow you to see graphs with metrics on different parts of the deployment. Not all of these are relevant, but these can be helpful in seeing CPU utilization in [RDS](https://aws.amazon.com/rds/) (CiviForm's PostgreSQL database) and [ECS](https://aws.amazon.com/ecs/) (server hosting), as well as requests to the [ALB](https://aws.amazon.com/elasticloadbalancing/application-load-balancer/) (load balancer), metrics about [S3](https://aws.amazon.com/s3/) (file storage), etc.
![](<../../.gitbook/assets/aws-cloudwatch-dashboards.png>)
```

File: docs/it-manual/sre-playbook/monitoring.md [hunk 2]
```diff
@@ -57,3 +60,68 @@ To import this pre-built dashboard:
...
+
+#### Alarms
+
+Some alarms are configured by default through terraform, which includes the following:
```
Comment (bion) [inline docs/it-manual/sre-playbook/monitoring.md:77]:
```suggestion
Some alarms are configured by default through the CiviForm deployment system, including the following: 
```

File: docs/it-manual/sre-playbook/monitoring.md [hunk 3]
```diff
@@ -57,3 +60,68 @@ To import this pre-built dashboard:
...
+  - Anomalous connection count alarm
+    - Related variable: `RDS_CREATE_ANOMALY_ALARM`
+
+For each of the RDS alarms, the variables `RDS_ALARM_EVALUATION_PERIOD` and `RDS_ALARM_STATISTIC_PERIOD` also apply.
+
+These alarms can be viewed through the AWS management console by clicking `All alarms` in the CloudWatch menu.
+
+The related variables can be added to the civiform_config.sh file of the forked civiform-deploy repository to customize the alarm settings or disable / enable certain alarms.
+
+If there are alarms that you wish to have, please let us know.
```
Comment (bion) [inline docs/it-manual/sre-playbook/monitoring.md:111]:
```suggestion
If there are other alarms that you wish to add, please let us know.
```

File: docs/it-manual/sre-playbook/monitoring.md [hunk 4]
```diff
@@ -57,3 +60,68 @@ To import this pre-built dashboard:
...
+
+### CloudWatch
+
+#### Dashboards
+
+CloudWatch has some default dashboards that allow you to see graphs with metrics on different parts of the deployment. Not all of these are relevant, but these can be helpful in seeing CPU utilization in RDS (the database) and Elastic Container Service, as well as requests to the Application ELB (load balancer), metrics about S3, etc.
+![](<../../.gitbook/assets/aws-cloudwatch-dashboards.png>)
+
+#### Alarms
+
```
Comment (bion) [inline docs/it-manual/sre-playbook/monitoring.md:76]: **suggestion (blocking):** Add a brief discussion of what an alarm is and what happens when an alarm goes off along with a link to the AWS docs

File: docs/it-manual/sre-playbook/monitoring.md [hunk 5]
```diff
@@ -57,3 +60,68 @@ To import this pre-built dashboard:
...
+- RDS:
+  - High CPU alarm
+    - Related variables: `RDS_CREATE_HIGH_CPU_ALARM`, `RDS_MAX_CPU_UTILIZATION_THRESHOLD`
+  - High disk queue depth alarm
+    - Related variables: `RDS_CREATE_HIGH_QUEUE_DEPTH_ALARM`, `RDS_DISK_QUEUE_DEPTH_HIGH_THRESHOLD`
+  - Low disk space alarm
+    - Related variables: `RDS_CREATE_LOW_DISK_SPACE_ALARM`, `RDS_DISK_FREE_STORAGE_LOW_THRESHOLD`
+  - Low freeable memory alarm
+    - Related variables: `RDS_CREATE_LOW_MEMORY_SPACE_ALARM`, `RDS_LOW_MEMORY_THRESHOLD`
+  - Maximum used transaction IDs alarm
```
Comment (bion) [inline docs/it-manual/sre-playbook/monitoring.md:92]: **suggestion (non-blocking):** add a parenthetical explaining what this is

File: docs/it-manual/sre-playbook/monitoring.md [hunk 6]
```diff
@@ -57,3 +60,68 @@ To import this pre-built dashboard:
...
+      - Related variable: `RDS_MAX_USED_TRANSACTION_IDS_HIGH_THRESHOLD`
+
+There are also the following alarms that can be enabled for RDS, but aren't created by default:
```
Comment (bion) [inline docs/it-manual/sre-playbook/monitoring.md:95]: **suggestion (blocking):** Mention how one enables these

File: docs/it-manual/sre-playbook/monitoring.md [hunk 7]
```diff
@@ -57,3 +60,68 @@ To import this pre-built dashboard:
...
+
+If there are alarms that you wish to have, please let us know.
+
+### RDS database metrics
+
+If you navigate to `RDS` in the AWS Console and click `Databases` in the navigation menu, you will see your database. In clicking on the database, you can see some basic metrics, like the current CPU percentage and activity. Additional metrics, which are the same as those in CloudWatch can be seen in the `Monitoring` section, and you can click other tabs (Logs, Configuration, etc.) to understand more about the Database configuration. 
+
+Database customization variables, including the one for the instance class and storage amount can be added to the civiform_config.sh file of the forked civiform-deploy repository to customize the configuration. 
+
+### Elastic Container Service (ECS) metrics
```
Comment (bion) [inline docs/it-manual/sre-playbook/monitoring.md:119]: **nitpick:** for consistency with "RDS" and the other acronyms we use
```suggestion
### ECS metrics
```

File: docs/it-manual/sre-playbook/monitoring.md [hunk 8]
```diff
@@ -57,3 +60,68 @@ To import this pre-built dashboard:
...
+
+If you navigate to `Elastic Container Service` in the AWS Console and click `Clusters` in the menu, you'll see the CiviForm cluster. When you click on the service, you can see health metrics (similar to those in CloudWatch).
+
+In the Configuration section of the service, you can see the Auto Scaling policies that are set up.
```
Comment (bion) [inline docs/it-manual/sre-playbook/monitoring.md:123]: **nitpick:** 
```suggestion
In the Configuration section of the service, you can see current Auto Scaling policies. 
```

**Review by bion — APPROVED** [review]

File: docs/it-manual/sre-playbook/monitoring.md (hunk 5 above)
Comment (dkatzz) [reply docs/it-manual/sre-playbook/monitoring.md]: This got removed when I removed the engine variable to disable the alarm, but I will add it back later and setup a boolean for this to be enabled / disabled as well.

File: docs/it-manual/sre-playbook/monitoring.md (hunk 4 above)
Comment (dkatzz) [reply docs/it-manual/sre-playbook/monitoring.md:76]: Updated.

File: docs/it-manual/sre-playbook/monitoring.md (hunk 6 above)
Comment (dkatzz) [reply docs/it-manual/sre-playbook/monitoring.md]: Updated.

**Pull request merged**